docs/spec_draft.md

```diff
@@ -522,6 +523,83 @@ it only exists to establish data dependencies from `result` to `inputs`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.all_reduce
+
+### Semantics
+
+Within each process group in the StableHLO grid, performs reduction, using
+`computations`, over the  values of the `operand` tensor from each process and
+produces a `result` tensor.
```

Comment:
Performs reduction, using `computations`, over the values of the `operand` tensor from each process in a `process_group` from the [StableHLO process grid](#parallel-execution) and produces a `result` tensor.

Reply:
feel free to omit `from the StableHLO process grid`, and link the process group to parallel semantics.

Reply:
The pros is inspired by suggestion in https://github.com/openxla/stablehlo/pull/503 (already incorporated in spec_draft.md).

I liked the idea of renaming "StableHLO grid" to "StableHLO process grid".
We can make this change all at once by opening a ticket.

Reply:
I also like the idea! Opened a ticket: #669.